numpy/fft/tests/test_helper.py

```diff
@@ -41,6 +41,68 @@ def test_axes_keyword(self):
         assert_array_almost_equal(fft.ifftshift(shifted, axes=0),
                 fft.ifftshift(shifted, axes=(0,)))
 
+        assert_array_almost_equal(fft.fftshift(freqs), shifted)
+        assert_array_almost_equal(fft.ifftshift(shifted), freqs)
+
+    def test_uneven_dims(self):
+        """ Test when 2D array has shape of 3x6 """
+        freqs = [[0, 1, 2], [3, 4, -4], [-3, -2, -1], [0, 1, 2], [3, 4, -4], [-3, -2, -1]]
+        shifted = [[2, 0, 1], [-4, 3, 4], [-1, -3, -2], [2, 0, 1], [-4, 3, 4], [-1, -3, -2]]
```

Comment:
nit: these tests would be more readable if you line-wrapped and aligned them. So
```
shifted = np.array([
    [ 2,  0,  1],
    [-4,  3,  4],
    [-1, -3, -2],
    [ 2,  0,  1],
    [-4,  3,  4],
    [-1, -3, -2]
])
```
and the same for `freqs`

Reply:
Actually, this is just a bad test, because the first three rows match the last three. Maybe just do `freq = np.arange(18).reshape((6, 3))`, and work out what `shifted` should look like.

Reply:
I will take a look during the weekend :)